File: backend/schemas/certification.py
"""
Pydantic schemas for certification API responses.
"""
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel

from models.shared_enums import CertificationStatus


class CertStandardResponse(BaseModel):
    """Cert standard metadata for API response."""

    id: str
    name: str
    required_score: float
    sessions_required: int
    weld_type: Optional[str] = None

    class Config:
        from_attributes = True


class CertificationStatusResponse(BaseModel):
    """Per-standard certification status for a welder."""

    cert_standard: CertStandardResponse
    status: CertificationStatus
    evaluated_at: datetime
    qualifying_sessions: int
    sessions_needed: int
    current_avg_score: Optional[float] = None
    sessions_to_target: Optional[int] = None
    qualifying_session_ids: Optional[List[str]] = None

    class Config:
        from_attributes = True


class WelderCertificationSummary(BaseModel):
    """Full certification summary for a welder across all standards."""

    welder_id: str
    certifications: List[CertificationStatusResponse]

File: my-app/src/types/certification.ts
/**
 * Types for certification standards and welder certification status.
 * Matches backend Pydantic models (snake_case).
 */
import type { CertificationStatus, WelderID } from "./shared";

export interface CertStandard {
  id: string;
  name: string;
  required_score: number;
  sessions_required: number;
  weld_type: string | null;
}

export interface CertificationStatusItem {
  cert_standard: CertStandard;
  status: CertificationStatus;
  evaluated_at: string;
  qualifying_sessions: number;
  sessions_needed: number;
  current_avg_score: number | null;
  sessions_to_target: number | null;
  qualifying_session_ids: string[] | null;
}

export interface WelderCertificationSummary {
  welder_id: WelderID;
  certifications: CertificationStatusItem[];
}
